<launch>
    <arg name="drone_id" default="0"/>
    <!-- size of map, change the size inflate x, y, z according to your application -->
    <arg name="map_size_x"/>
    <arg name="map_size_y"/>
    <arg name="map_size_z"/>
    <!-- Input/output topics -->
    <arg name="command_topic" default="/mavros/setpoint_raw/local" />
    <arg name="odometry_topic" default="/mavros/local_position/odom" />
    <arg name="pose_topic" default="/mavros/local_position/pose" />
    <arg name="drone_origin_frame" default="world" />
    
    <!-- rosBag file path -->
    <arg name="bag_file_prefix" default="~/gestelt/data/flight_data" />
    
    <!-- Trajectory server -->
    <node pkg="trajectory_server" name="trajectory_server" type="traj_server_node" output="screen">
        <param name="drone_id" value="$(arg drone_id)"/>
        <param name="origin_frame" value="$(arg drone_origin_frame)"/>

        <!-- Operational params -->
        <param name="traj_server/takeoff_height" value="1" type="double"/>

        <!-- Debug display params -->
        <param name="traj_server/uav_pose_history_size" value="5000" type="int"/>

        <!-- Safety bounding box, any commands exceeding this limit will result in the drone stopping-->
        <!-- This is relative to uav origin frame -->
        <param name="traj_server/enable_safety_box" value="false" type="bool"/>
        <param name="traj_server/safety_box/min_x" value="-6.0" type="double"/>
        <param name="traj_server/safety_box/min_y" value="-6.0" type="double"/>
        <param name="traj_server/safety_box/min_z" value="-1.5" type="double"/>
        <param name="traj_server/safety_box/max_x" value="6.0" type="double"/>
        <param name="traj_server/safety_box/max_y" value="6.0" type="double"/>
        <param name="traj_server/safety_box/max_z" value="5.0" type="double"/>

        <!-- Subscribed Topics -->
        <!-- <remap from="/planner/trajectory" to="/planner/trajectory"/> -->
        <!-- <remap from="/mavros/state" to="/mavros/state"/>
        <remap from="/mavros/local_position/pose" to="$(arg pose_topic)"/>
        <remap from="/mavros/local_position/odom" to="$(arg odometry_topic)"/> -->
 
        <!-- Published Topics -->
        <!-- <remap from="/mavros/setpoint_raw/local" to="$(arg command_topic)"/>
        <remap from="/uav_path_trajectory" to="/uav_path_trajectory"/> -->

        <!-- Service clients -->
        <!-- <remap from="/mavros/cmd/arming" to="/mavros/cmd/arming"/>
        <remap from="/mavros/set_mode" to="/mavros/set_mode"/> -->

        <!-- Planner Heartbeat -->
        <!-- <param name="traj_server/planner_heartbeat_timeout" value="0.5" type="double"/> -->
        <param name="traj_server/ignore_planner_heartbeat" value="true" type="bool"/>

        <!-- Frequency for timers -->
        <!-- <param name="traj_server/pub_cmd_freq" value="25.0" type="double"/>
        <param name="traj_server/state_machine_tick_freq" value="50.0" type="double"/>
        <param name="traj_server/debug_freq" value="10.0" type="double"/> -->

    </node>
    
    <!-- <node name="record_bag" pkg="rosbag" type="record" args="-a -d 60" output="screen">
        <param name="bag_filename" value="$(arg bag_file_prefix).bag" />
    </node> -->
  </launch>
